doc/sphinx/source/glossary.rst

```diff
@@ -120,5 +120,42 @@ Glossary
    export
    
       The serialized version of a trained model that can be passed to
-      one of the nGraph backends for computation.      
+      one of the nGraph backends for computation.
 
+   NN
+
+      See :term:`ANN`.
+
+   ANN
+
+      :abbr:`ANN (Artificial Neural Network)` is an acronym for "Artificial 
+      Neural Network". ANN models are used to simulate possible combinations of 
+      binary logic processing and multi-layer (multi-dimensional) paths through 
+      which a :term:`data-flow graph` may be mapped or computed. An ANN does not 
+      have centralized storage; rather, an ANN manifests as information stored 
+      as patterns throughout the network structure. ANNs may be **Recurrent** 
```

Comment:
Recurrent networks are just ones that unroll along some underlying structure in the input; typically a sequence or a (parse) tree.
Residual neural networks have a feedback mechanism (the residual); for example resnet.
Feed forward/backwards is a completely different thing having to do with whether you are trying to reduce a loss or not, where the loss is how much your results differ from the results you want; the feed backwards is used to compute how to adjust the weights to lower the loss.

Reply:
Thanks, I read a bunch of academic papers from the late 90's to get some of these defs, since I thought it would make sense to use the established lexicology, but maybe it has changed ... ?

Reply:
Adding the "Artificial" part anyhow.  And, WRT to "Recurrent" being anything with a loopback, I read a few articles and looked up an old slack thread where @aprocter tried to explain, and kind of merged them to get this.  I am trying to get the docs prepped for some of the dynamic graph docs we will be working on soon, since i am thinking these kinds of RNNs will need to be explained thoroughly regarding how they are different from each other.

Reply:
TF is putting the dynamic in the graph, rather than dealing with it in python. This gives more opportunity to the compiler. I'm in the process of grokking what they are doing right now.
https://arxiv.org/abs/1805.01772